src/test/java/org/openrewrite/staticanalysis/UnnecessaryReturnAsLastStatementTest.java

```diff
@@ -0,0 +1,123 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.staticanalysis;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+class UnnecessaryReturnAsLastStatementTest implements RewriteTest {
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipe(new UnnecessaryReturnAsLastStatement());
+    }
+
+    @Test
+    @DocumentExample
+    void simpleReturn() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              class Hello {
+                  void world() {
+                      System.out.println("Hello world");
+                      return;
+                  }
+              }
+              """,
+            """
+              class Hello {
+                  void world() {
+                      System.out.println("Hello world");
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void ifBranches() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              class Hello {
+                  void world(int i) {
+                      if (i > 0) {
+                        System.out.println("Positive");
+                        return;
+                      } else {
+                        System.out.println("Zero or negative");
+                        return;
+                      }
+                  }
+              }
+              """,
+            """
+              class Hello {
+                  void world(int i) {
+                      if (i > 0) {
+                        System.out.println("Positive");
+                      } else {
+                        System.out.println("Zero or negative");
+                      }
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void ifIsNotTheLast() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              class Hello {
+                  void world(int i) {
+                      if (i > 0) {
+                        System.out.println("Positive");
+                        return;
+                      } else {
+                        System.out.println("Zero or negative");
+                      }
+                      System.out.println("Some extra logic");
+                  }
+              }
+              """));
+    }
+
+    @Test
+    void notChangingNonVoidMethods() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              class Hello {
+                  int world(int i) {
+                      return i + 436;
+                  }
+              }
+              """));
+    }
+
```

Comment:
It might be good to add a test case similar to what we have in https://github.com/openrewrite/rewrite-static-analysis/commit/f1ce05b697a8787d77bfb001476495450cc30959 for a nested `J.NewClass` or `J.Lambda`.

Reply:
3325148629f54c5362a96df3bc59f4418051f123